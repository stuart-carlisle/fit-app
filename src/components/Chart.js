import React from 'react'
import { Line } from 'react-chartjs-2'

export default Chart = ({ chartData, chartDataMobile, chartDataTablet }) =>{ 
    
    const chart = (<Line
        data={chartData.data}
        height={200}
        options={chartData.options}
        />)

    const chartMobile = (<Line
        data={chartDataMobile.data}
        height={300}
        options={chartDataMobile.options}
        />)
    
    const chartTablet = (<Line
        data={chartDataTablet.data}
        height={200}
        options={chartDataTablet.options}
        />)
    
    const jsx= chartData ? chart :(<p>No chart data</p>)
    const jsxMobile= chartDataMobile ? chartMobile :(<p>No chart data</p>)
    const jsxTablet= chartDataTablet ? chartTablet :(<p>No chart data</p>)

    return(
        <div>
        <div className="content-container content-container--chart show-chart-large">
            {jsx}
        </div>
        <div className="content-container content-container--chart show-chart-small">
            {jsxMobile}
        </div>
        <div className="content-container content-container--chart show-chart-medium">
            {jsxTablet}
        </div>
        </div>
    )   
}
