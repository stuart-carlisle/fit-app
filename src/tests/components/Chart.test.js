import React from 'react'
import { shallow } from 'enzyme'
import  Chart   from '../../components/Chart'
import chartData from '../fixtures/chart'

test('Should render the chart component with valid data',()=>{  
    const wrapper = shallow(<Chart chartData={chartData} chartDataMobile={chartData} chartDataTablet={chartData} />)
    expect(wrapper).toMatchSnapshot()
    
})