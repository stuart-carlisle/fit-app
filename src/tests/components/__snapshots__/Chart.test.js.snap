// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the chart component with valid data 1`] = `
<div>
  <div
    className="content-container content-container--chart show-chart-large"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "borderColor": "rgba(68, 114, 199, 0.6)",
              "data": Array [
                Object {
                  "x": 0,
                  "y": 487,
                },
                Object {
                  "x": -172800000,
                  "y": 342,
                },
              ],
              "fill": false,
              "label": "Calories",
              "lineTension": 0.3,
              "pointBackgroundColor": Array [
                "rgba(255, 99, 132, 0.6)",
                "rgba(54, 162, 235, 0.6)",
                "rgba(255, 206, 86, 0.6)",
                "rgba(75, 192, 192, 0.6)",
                "rgba(153, 102, 255, 0.6)",
                "rgba(255, 159, 64, 0.6)",
                "rgba(255, 99, 132, 0.6)",
                "rgba(255, 99, 132, 0.6)",
              ],
            },
          ],
        }
      }
      height={200}
      options={
        Object {
          "legend": Object {
            "display": false,
            "position": "right",
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "DATE",
                },
                "ticks": Object {
                  "callback": [Function],
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                  "max": -3600000,
                  "min": -522000000,
                },
                "time": Object {
                  "unit": "day",
                },
                "type": "linear",
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "CALORIES PER DAY",
                },
                "ticks": Object {
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "fontColor": "#4472C7",
            "fontFamily": "Agency FB",
            "fontSize": 25,
            "text": "CALORIE TRACKER FOR THE LAST WEEK",
          },
        }
      }
    />
  </div>
  <div
    className="content-container content-container--chart show-chart-small"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "borderColor": "rgba(68, 114, 199, 0.6)",
              "data": Array [
                Object {
                  "x": 0,
                  "y": 487,
                },
                Object {
                  "x": -172800000,
                  "y": 342,
                },
              ],
              "fill": false,
              "label": "Calories",
              "lineTension": 0.3,
              "pointBackgroundColor": Array [
                "rgba(255, 99, 132, 0.6)",
                "rgba(54, 162, 235, 0.6)",
                "rgba(255, 206, 86, 0.6)",
                "rgba(75, 192, 192, 0.6)",
                "rgba(153, 102, 255, 0.6)",
                "rgba(255, 159, 64, 0.6)",
                "rgba(255, 99, 132, 0.6)",
                "rgba(255, 99, 132, 0.6)",
              ],
            },
          ],
        }
      }
      height={300}
      options={
        Object {
          "legend": Object {
            "display": false,
            "position": "right",
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "DATE",
                },
                "ticks": Object {
                  "callback": [Function],
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                  "max": -3600000,
                  "min": -522000000,
                },
                "time": Object {
                  "unit": "day",
                },
                "type": "linear",
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "CALORIES PER DAY",
                },
                "ticks": Object {
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "fontColor": "#4472C7",
            "fontFamily": "Agency FB",
            "fontSize": 25,
            "text": "CALORIE TRACKER FOR THE LAST WEEK",
          },
        }
      }
    />
  </div>
  <div
    className="content-container content-container--chart show-chart-medium"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "borderColor": "rgba(68, 114, 199, 0.6)",
              "data": Array [
                Object {
                  "x": 0,
                  "y": 487,
                },
                Object {
                  "x": -172800000,
                  "y": 342,
                },
              ],
              "fill": false,
              "label": "Calories",
              "lineTension": 0.3,
              "pointBackgroundColor": Array [
                "rgba(255, 99, 132, 0.6)",
                "rgba(54, 162, 235, 0.6)",
                "rgba(255, 206, 86, 0.6)",
                "rgba(75, 192, 192, 0.6)",
                "rgba(153, 102, 255, 0.6)",
                "rgba(255, 159, 64, 0.6)",
                "rgba(255, 99, 132, 0.6)",
                "rgba(255, 99, 132, 0.6)",
              ],
            },
          ],
        }
      }
      height={200}
      options={
        Object {
          "legend": Object {
            "display": false,
            "position": "right",
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "DATE",
                },
                "ticks": Object {
                  "callback": [Function],
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                  "max": -3600000,
                  "min": -522000000,
                },
                "time": Object {
                  "unit": "day",
                },
                "type": "linear",
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 20,
                  "labelString": "CALORIES PER DAY",
                },
                "ticks": Object {
                  "fontColor": "#4472C7",
                  "fontFamily": "Agency FB",
                  "fontSize": 15,
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "fontColor": "#4472C7",
            "fontFamily": "Agency FB",
            "fontSize": 25,
            "text": "CALORIE TRACKER FOR THE LAST WEEK",
          },
        }
      }
    />
  </div>
</div>
`;
